modules/pubmaticBidAdapter.js

```diff
@@ -818,6 +849,7 @@ export const spec = {
       if (impObj) {
         payload.imp.push(impObj);
       }
+      schainConfig = schainConfig || bid.schain;
```

Comment:
This line seems a bit strange to me.

According to the implementation of the `copySchainObjectInAdunits(..)` function, it seems like you would always have one of the following two cases:

- The Supply Chain object has been injected into _all_ bids
- The Supply Chain object has been injected into _none of_ the bids

However, this specific line of code suggests that there could be scenarios where some bids of a Supply Chain, and some do not.  Furthermore, if a Bid is missing a Supply Chain, then it's somehow acceptable to copy the Supply Chain from the previous Bid...?

This is also making me question whether or not the Supply Chain should actually be injected into individual bids.  It seems like "Supply Chain" is a property of the request as a whole.

Reply:
Coming back to this: it seems like the Supply Chain should be injected into the `bidderRequest` rather than the individual `validBidRequests`.  This would eliminate the need for this strange line of code.

Reply:
(To be fair: I'm not actually super familiar w/ the Prebid.js API since I haven't worked on this codebase before.  So, `bidderRequest` might not be serving the purpose that I think it serves.  However, my main point is that it seems like the Supply Chain should be a property of the request/transaction as a whole, rather than a property of the individual Bids)

Reply:
I also agree that we should not duplicate the schain data in bids.
I am not sure how we can pass schain data in bidderRequest object through an optional module, this can be done easily from adapterManager.